packages/plugins/i18n/server/src/services/localizations.ts

```diff
@@ -1,40 +1,49 @@
 import { isEmpty } from 'lodash/fp';
 
-import { async } from '@strapi/utils';
 import { getService } from '../utils';
 
-const isDialectMySQL = () => strapi.db.dialect.client === 'mysql';
-
 /**
  * Update non localized fields of all the related localizations of an entry with the entry values
- * @param {Object} entry entry to update
- * @param {Object} options
- * @param {Object} options.model corresponding model
  */
-const syncNonLocalizedAttributes = async (entry: any, { model }: any) => {
+const syncNonLocalizedAttributes = async (sourceEntry: any, model: any) => {
   const { copyNonLocalizedAttributes } = getService('content-types');
 
-  if (Array.isArray(entry?.localizations)) {
-    const nonLocalizedAttributes = copyNonLocalizedAttributes(model, entry);
-
-    if (isEmpty(nonLocalizedAttributes)) {
-      return;
-    }
-
-    const updateLocalization = (id: any) => {
-      return strapi.entityService.update(model.uid, id, { data: nonLocalizedAttributes });
-    };
-
-    // MySQL/MariaDB can cause deadlocks here if concurrency higher than 1
-    // TODO: use a transaction to avoid deadlocks
-    await async.map(
-      entry.localizations,
-      (localization: any) => updateLocalization(localization.id),
-      {
-        concurrency: isDialectMySQL() && !strapi.db.inTransaction() ? 1 : Infinity,
-      }
-    );
+  const nonLocalizedAttributes = copyNonLocalizedAttributes(model, sourceEntry);
+  if (isEmpty(nonLocalizedAttributes)) {
+    return;
+  }
+
+  const uid = model.uid;
+  const documentId = sourceEntry.documentId;
+  const status = sourceEntry?.publishedAt ? 'published' : 'draft';
+
+  // Find all the entries that need to be updated
+  // Every other entry of the document in the same status but a different locale
+  const entriesToUpdate = await strapi.db.query(uid).findMany({
+    where: {
+      documentId,
+      publishedAt: status === 'published' ? { $ne: null } : null,
+      locale: { $ne: sourceEntry.locale },
+    },
+    select: ['locale', 'id'],
+  });
+
+  for (const entry of entriesToUpdate) {
+    await strapi.documents(uid).updateComponents(uid, entry, nonLocalizedAttributes as any);
   }
+  const entryData = await strapi.documents(uid).omitComponentData(model, nonLocalizedAttributes);
+
+  // Update every other locale entry of this documentId in the same status
+  // We need to support both statuses incase we are working with a content type
+  // without draft and publish enabled
+  await strapi.db.query(uid).updateMany({
+    where: {
+      documentId,
+      publishedAt: status === 'published' ? { $ne: null } : null,
+      locale: { $in: entriesToUpdate.map((entry) => entry.locale) },
+    },
+    data: entryData,
+  });
```

Comment:
probably we were not doing it before,but with this approach we will not be triggering any webhook / audit log.
Not for this PR , but as an idea for the future, what if we use the document service methods to update each local individually rather than calling the db directly?

For performance, we could even make this async.

wdyt @jhoward1994 @alexandrebodin ?

Reply:
I'm not sure if we want to trigger webhooks 🤔 need to discuss with product I think

Reply:
I discussed this with Paridhi and she is happy with not dispatching webhooks for
the locales modified as a side effect of a non localized field